nixos/modules/services/misc/snapper.nix

```diff
@@ -155,10 +163,25 @@ in
     systemd.timers.snapper-cleanup = {
       description = "Cleanup of Snapper Snapshots";
       inherit documentation;
-      wantedBy = [ "basic.target" ];
+      wantedBy = [ "timers.target" ];
       timerConfig.OnBootSec = "10m";
       timerConfig.OnUnitActiveSec = cfg.cleanupInterval;
     };
+
+    systemd.services.snapper-boot = lib.optionalAttrs cfg.snapshotOnBoot {
+      description = "Take snapper snapshot of root on boot";
+      inherit documentation;
+      serviceConfig.ExecStart = "${pkgs.snapper}/bin/snapper --config root create --cleanup-algorithm number --description boot";
+      serviceConfig.type = "oneshot";
+      unitConfig.ConditionPathExists = "/etc/snapper/configs/root";
+    };
+
+    systemd.timers.snapper-boot = lib.optionalAttrs cfg.snapshotOnBoot {
```

Comment:
All services should depend `local-fs.target` to ensure the file system is actually mounted?